gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/components/SelectableAvatar.kt

```diff
@@ -21,66 +25,107 @@ import androidx.compose.ui.unit.dp
 ) {
     val cornerRadius = 8.dp
-    Box(modifier = modifier.aspectRatio(1f)) {
+    Box(
+        modifier = modifier
+            .aspectRatio(1f)
+            .then(
+                if (isSelected) {
+                    Modifier.border(4.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(cornerRadius))
+                } else {
+                    Modifier.border(
+                        1.dp,
+                        MaterialTheme.colorScheme.surfaceDim,
+                        RoundedCornerShape(cornerRadius),
+                    )
+                },
+            )
+            .clickable {
+                onAvatarClicked()
+            },
+    ) {
         AsyncImage(
             model = imageUrl,
             contentDescription = stringResource(id = R.string.selectable_avatar_content_description),
             modifier = Modifier
                 .fillMaxSize()
-                .clip(RoundedCornerShape(cornerRadius))
-                .then(
-                    if (isSelected) {
-                        Modifier.border(4.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(cornerRadius))
-                    } else {
-                        Modifier.border(
-                            1.dp,
-                            MaterialTheme.colorScheme.surfaceDim,
-                            RoundedCornerShape(cornerRadius),
-                        )
-                    },
-                )
-                .clickable {
-                    onAvatarClicked()
-                },
+                .clip(RoundedCornerShape(cornerRadius)),
         )
-        if (isLoading) {
-            Box(
-                modifier = modifier
-                    .fillMaxSize()
-                    .background(
-                        color = Color.Black.copy(alpha = 0.3f),
-                        shape = RoundedCornerShape(cornerRadius),
-                    ),
-            ) {
-                CircularProgressIndicator(
-                    modifier = Modifier
-                        .align(Alignment.Center)
-                        .size(20.dp),
-                    strokeWidth = 2.dp,
-                )
-            }
+        when (loadingState) {
+            AvatarLoadingState.None -> Unit
+            AvatarLoadingState.Loading -> LoadingOverlay()
+            is AvatarLoadingState.Failure -> FailureOverlay()
         }
     }
 }
 
+@Composable
+private fun LoadingOverlay(modifier: Modifier = Modifier) {
+    Overlay(modifier) {
+        CircularProgressIndicator(
+            modifier = Modifier
+                .align(Alignment.Center)
+                .size(20.dp),
+            strokeWidth = 2.dp,
+        )
+    }
+}
+
+@Composable
+private fun FailureOverlay(modifier: Modifier = Modifier) {
+    Overlay(modifier) {
+        Icon(
+            imageVector = Icons.Rounded.Warning,
+            contentDescription = stringResource(R.string.failed_to_load_avatar_content_description),
+            tint = Color.White,
```

Comment:
⛏ If we use hardcoded colors and someone overrides the `GravatarTheme` to use a different color palette, these colors won't change. In this case, the `White` is hardcoded in the overlay, which has a hardcoded color, so at least it should always look good.

Reply:
I've used hardcoded colors here on purpose as we display the icon on top of an image - so we don't have a variable (light/dark) background here. 

What color from the theme should I use instead?

Reply:
It's not about light/dark themes. We let developers fully [customize colors](https://github.com/Automattic/Gravatar-SDK-Android/blob/trunk/docs/get-started/get-started.md#customization) inside the Gravatar SDK via GravataTheme.

For example, with just a few lines, I can modify how the demo-app shows the QE bottomsheet:

```kotlin
CompositionLocalProvider(LocalGravatarTheme provides object : GravatarTheme {
    override val colorScheme: ColorScheme
        @Composable
        get() = MaterialTheme.colorScheme.copy(surface = Color.Red, primary = Color.Green)
}) {

```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/f98812f8-bdb9-4dfc-a176-806fbbf577a6">

If we hardcode the colors here, this will be the only place where we won't use the theme-provided colors.

> What color from the theme should I use instead?

That's a good question. `MaterialTheme.colorScheme.error` sounds reasonable, but maybe it doesn't look as good as the white you've picked.

<img width="250" alt="image" src="https://github.com/user-attachments/assets/2edb1f88-0f1c-4dd6-9c86-9ee98350ea5d">

Reply:
>It's not about light/dark themes. We let developers fully [customize colors](https://github.com/Automattic/Gravatar-SDK-Android/blob/trunk/docs/get-started/get-started.md#customization) inside the Gravatar SDK via GravataTheme.


You're right. Tbh, I think we shouldn't allow that in the QE. We should for `:ui` components, but not in the Editor. 

QuickEditor is not a component devs will embed into their screens directly - it's an in-app Gravatar app pretty much. So you launch it within the app but it should be clear you are in the "Gravatar" world in that bottom sheet.

Reply:
> I think we shouldn't allow that in the QE. We should for :ui components, but not in the Editor.
> 
> So you launch it within the app but it should be clear you are in the "Gravatar" world

I completely see the point of that. However, we had similar discussions with the UI components, and in the end, if someone decides to customize the Editor, they can do it by forking the repo and applying a different theme. That might be a good discussion for the whole team.